src/pages/index.js

```diff
@@ -54,43 +58,246 @@ const addFormValidator = new FormValidator(
-  const card = new Card(cardData, "#card-template", handleImageClick);
+  const sanitizedLink =
+    cardData.link.replace(/["'\s]+$/, "").trim() ||
+    "https://placehold.co/150x150";
+  const card = new Card(
+    {
+      name: cardData.name,
+      link: sanitizedLink,
+      _id: cardData._id || null,
+    },
+    "#card-template",
+    handleImageClick,
+    handleDeleteClick,
+    handleLikeClick
+  );
+
   const element = card.getView();
+  if (!cardData._id) {
+    console.warn("Card created with no _id, deletion might fail:", cardData);
+  } else {
+    const initialIsLiked =
+      cardData.isLiked !== undefined ? cardData.isLiked : false;
+    likeStatuses.set(cardData._id, initialIsLiked);
+    const likeButton = element.querySelector(".card__like-button");
+    if (likeButton) {
+      likeButton.classList.toggle("card__like-button_active", initialIsLiked);
+    }
+  }
   return element;
 }
 
 function handleProfileFormSubmit(data) {
-  userInfo.setUserInfo({
-    name: data["edit-profile"],
-    job: data["profile-description"],
-  });
-  profileEditPopup.close();
+  const saveButton = profileEditForm.querySelector(".modal__save-button");
+  saveButton.textContent = "Saving...";
+  saveButton.classList.add("loading");
+  saveButton.disabled = true;
+  api
+    .updateUserInfo(data["edit-profile"], data["profile-description"])
+    .then((updatedInfo) => {
+      userInfo.setUserInfo({
+        name: updatedInfo.name,
+        about: updatedInfo.about,
+        avatar: updatedInfo.avatar,
+      });
+      profileEditPopup.close();
+    })
+    .catch((err) => {
+      console.error("Error updating user info:", err);
+    })
+    .finally(() => {
+      saveButton.textContent = "Save";
+      saveButton.classList.remove("loading");
+      saveButton.disabled = false;
+    });
 }
 
 function handleAddCardFormSubmit(data) {
-  const newCardElement = createCard({ name: data.title, link: data.url });
-  section.addItem(newCardElement);
-  addCardFormElement.reset();
-  addFormValidator.disableButton();
-  addCardPopup.close();
+  const sanitizedUrl = data.url.replace(/["'\s]+$/, "").trim();
+  const saveButton = addCardFormElement.querySelector(".modal__save-button");
+  saveButton.textContent = "Saving...";
+  saveButton.classList.add("loading");
+  saveButton.disabled = true;
+  api
+    .addCard(data.title, sanitizedUrl)
+    .then((newCardInfo) => {
+      const newCardElement = createCard({
+        name: newCardInfo.name,
+        link: newCardInfo.link,
+        _id: newCardInfo._id,
+      });
+      section.addItem(newCardElement);
+      addCardFormElement.reset();
+      addFormValidator.resetValidation();
+      addCardPopup.close();
+    })
+    .catch((err) => {
+      console.error("Error adding new card:", err);
+    })
+    .finally(() => {
+      saveButton.textContent = "Save";
+      saveButton.classList.remove("loading");
+      saveButton.disabled = false;
+      addCardPopup.close();
```

Comment:
You need to close modals only in `then` blocks after a **successful response** from the server so that when there is an error, it would be opened (the inputs will not be cleared and the user could send the data again to the server), and if you change the button text, you could see the change as well. **Please, correct it everywhere in the code**